suite-native/tokens/src/tokensSelectors.ts

```diff
@@ -47,63 +54,52 @@ export const selectAccountTokenInfo = memoizeWithArgs(
     },
-    // 100 is a reasonable size for the cache
-    { size: 100 },
 );
 
-export const selectAccountTokenSymbol = (
-    state: AccountsRootState,
-    accountKey?: AccountKey,
-    tokenAddress?: TokenAddress,
-): TokenSymbol | null => {
-    const tokenInfo = selectAccountTokenInfo(state, accountKey, tokenAddress);
-
-    if (!tokenInfo) {
-        return null;
-    }
-
-    // FIXME: This is the only place in the codebase where we change case of token symbol.
-    // The `toUpperCase()` operation is necessary because we are receiving wrongly formatted token symbol from connect.
-    // Can be removed at the moment when desktop issue https://github.com/trezor/trezor-suite/issues/8037 is resolved.
-    return tokenInfo.symbol.toUpperCase() as TokenSymbol;
-};
-
-export const selectAccountTokenBalance = (
-    state: AccountsRootState,
-    accountKey?: AccountKey,
-    tokenAddress?: TokenAddress,
-): string | null => {
-    const tokenInfo = selectAccountTokenInfo(state, accountKey, tokenAddress);
+export const selectAccountTokenSymbol = createAppSelector(
+    [selectAccountTokenInfo],
+    (tokenInfo): TokenSymbol | null => {
+        if (!tokenInfo) {
+            return null;
+        }
 
-    if (!tokenInfo) {
-        return null;
-    }
+        // FIXME: This is the only place in the codebase where we change case of token symbol.
+        // The `toUpperCase()` operation is necessary because we are receiving wrongly formatted token symbol from connect.
+        // Can be removed at the moment when desktop issue https://github.com/trezor/trezor-suite/issues/8037 is resolved.
+        return tokenInfo.symbol.toUpperCase() as TokenSymbol;
+    },
+);
 
-    return tokenInfo.balance ?? null;
-};
+export const selectAccountTokenBalance = createAppSelector(
+    [selectAccountTokenInfo],
+    (tokenInfo): string | null => {
+        if (!tokenInfo) {
+            return null;
+        }
 
-export const selectAccountTokenDecimals = (
-    state: AccountsRootState,
-    accountKey?: AccountKey,
-    tokenAddress?: TokenAddress,
-): number | null => {
-    const tokenInfo = selectAccountTokenInfo(state, accountKey, tokenAddress);
+        return tokenInfo.balance ?? null;
+    },
+);
 
-    if (!tokenInfo) {
-        return null;
-    }
+export const selectAccountTokenDecimals = createAppSelector(
+    [selectAccountTokenInfo],
+    (tokenInfo): number | null => {
+        if (!tokenInfo) {
+            return null;
+        }
 
-    return tokenInfo.decimals ?? null;
-};
+        return tokenInfo.decimals ?? null;
+    },
+);
 
-export const selectAccountTokenTransactions = memoizeWithArgs(
-    (
-        state: TransactionsRootState,
-        accountKey: AccountKey,
-        tokenAddress: TokenAddress,
-    ): WalletAccountTransaction[] =>
+export const selectAccountTokenTransactions = createAppSelector(
+    [
+        selectAccountTransactions,
+        (_state, _accountKey: AccountKey, tokenAddress: TokenAddress) => tokenAddress,
```

Comment:
Damn, this is so ugly that you have to pass a function instead of a simple argument value 🥴. But it is how it is

Reply:
It's even worse if you have multiple nested selectors with different params :D